includes/admin/import/class-batch-import-downloads.php

```diff
@@ -0,0 +1,459 @@
+		}
+
+		return $percentage;
+	}
+
+	private function str_to_array( $str = '' ) {
+
+		// Look for standard delimiters
+		if( false !== strpos( $str, '|' ) ) {
+
+			$delimiter = '|';
+
+		} elseif( false !== strpos( $str, ',' ) ) {
+
+			$delimiter = ',';
+
+		} elseif( false !== strpos( $str, ';' ) ) {
+
+			$delimiter = ';';
+
+		}
+
+		if( ! empty( $delimiter ) ) {
+
+			$array = (array) explode( $delimiter, $str );
+
+			return array_map( 'trim', $array );
+
+		}
+
+		return array();
+
+	}
+
+	private function set_price( $download_id = 0, $price = '' ) {
+
+		if( is_numeric( $price ) ) {
+
+			update_post_meta( $download_id, 'edd_price', edd_sanitize_amount( $price ) );
+
+		} else {
+
+			$prices = $this->str_to_array( $price );
+
+			if( ! empty( $prices ) ) {
+
+				$variable_prices = array();
+				foreach( $prices as $price ) {
+
+					// See if this matches the EDD Download export for variable prices
+					if( false !== strpos( $price, ':' ) ) {
+
+						$price = array_map( 'trim', explode( ':', $price ) );
+
+						$variable_prices[] = array( 'name' => $price[0], 'amount' => $price[1] );
+
+					}
+
+				}
+
+				update_post_meta( $download_id, '_variable_pricing', 1 );
+				update_post_meta( $download_id, 'edd_variable_prices', $variable_prices );
+
+			}
+
+		}
+
+	}
+
+	private function set_files( $download_id = 0, $files = array() ) {
+
+		if( ! empty( $files ) ) {
+
+			$download_files = array();
+			foreach( $files as $file ) {
+
+				$download_files[] = array( 'file' => $file, 'name' => basename( $file ) );
+
+			}
+
+			update_post_meta( $download_id, 'edd_download_files', $download_files );
+
+		}
+
+	}
+
+	private function set_image( $download_id = 0, $image = '' ) {
+
+		$is_url   = false !== filter_var( $image, FILTER_VALIDATE_URL );
+		$is_local = $is_url && false !== strpos( $image, site_url() );
+		$ext      = edd_get_file_extension( $image );
+
+		if( $is_url && $is_local ) {
+
+			// Image given by URL, see if we have an attachment already
+			$attachment_id = attachment_url_to_postid( $image );
+
+		} elseif( $is_url ) {
+
+			if( ! function_exists( 'media_sideload_image' ) ) {
+
+				require_once( ABSPATH . 'wp-admin/includes/file.php' );
+
+			}
+
+			// Image given by external URL
+			$url = media_sideload_image( $image, $download_id, '', 'src' );
+
+			if( ! is_wp_error( $url ) ) {
+
+				$attachment_id = attachment_url_to_postid( $url );
+
+			}
+
+
+		} elseif( false === strpos( $image, '/' ) && edd_get_file_extension( $image ) ) {
+
+			// Image given by name only
+
+			$upload_dir = wp_upload_dir();
+
+			if( file_exists( trailingslashit( $upload_dir['path'] ) . $image ) ) {
+
+				// Look in current upload directory first
+				$file = trailingslashit( $upload_dir['path'] ) . $image;
+
+			} else {
+
+				// Now look through year/month sub folders of upload directory for files with our image's same extension
+				$files = glob( $upload_dir['basedir'] . '/*/*/*{' . $ext . '}', GLOB_BRACE );
```

Comment:
When split into subfolders, files are never stored in year only folders. There is always a month.